cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/BatchPartitionConsumer.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset.lib;
+
+import co.cask.cdap.api.annotation.Beta;
+import co.cask.cdap.api.data.DatasetContext;
+import com.google.common.collect.Maps;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * A utility class to assist an incremental consumer of {@link Partition}s of a {@link PartitionedFileSet}. The
+ * implementing class is required to handle persistence of the consumption state via the abstract readBytes and
+ * writeBytes methods.
+ */
+@Beta
+public abstract class BatchPartitionConsumer {
+  private PartitionConsumerState finalConsumerState;
+
+  /**
+   * @return the serialized bytes of the state of the partition consuming MapReduce; return null to indicate a fresh
```

Comment:
Ah, it originally was implemented to be, but then tried to make it more generic (so also can be used from spark in the future).
I will update the javadoc.